### PR TITLE
Add doc to handle lexical error in error recovery

### PR DESCRIPTION
**Pull request by chanbengz**
Follow #992. I found a weird exit of parser when inputing an invalid token to trigger lexical error, where I expected error recovery should work. After discussing with @dburgener, we decide to address this issue temporarily in book. 

Since it's a minor change in user's code, I think it'll be unnecessary to provide an example code here. If needed, I have a copy of the example code on hand. Hope it could help.

### CODE REVIEW

**Comment by Pat-Lafon** [comment]
The github actions workflow suggests there are a few lint nits related to trailing spaces and such that should be an easy fix.

**Comment by chanbengz** [comment]
Oh, I wasn't aware of that lint check. Should be okay now.

**Comment by chanbengz** [comment]
Ah, sorry. Miss somewhere, and it got fixed now.

**Comment by dburgener** [comment]
> Ah, sorry. Miss somewhere, and it got fixed now.

No worries.  It's annoying for first time contributors, because you have to wait for a maintainer to hit the "approve workflow" button to get the workflow to run, so the iteration can be tricky.  If you want to check things yourself locally, `sudo gem install mdl` and `mdl doc` are the instructions run by the CI (they assume you already have a ruby environment set up though...)

I haven't had a chance to look at this yet.  It's a busy week with the Thanksgiving holiday in the US.  I'm hoping to get to it before the holiday, but I'm not sure yet.  If I don't get to it this week, I'll look early next week.

**Comment by dburgener** [comment]
latest CI failure is something we're seeing broadly due to a dependency, and not an issue with this PR.

**Comment by chanbengz** [comment]
Gotcha, thanks for the remind. I squashed the fixes into one commit and it won't change other things.

**Comment by dburgener** [comment]
Thanks for the PR!

There were a few inaccuracies in the problem description and the situation with the built-in lexer, so I pushed to try to clarify all that, and fix a few grammar issues.

**Review by dburgener — APPROVED** [review]

**Comment by dburgener** [comment]
Merged.  Thanks for the submission!

**Pull request merged**
